site/app/controllers/student/SubmissionController.php

```diff
@@ -1020,6 +1020,63 @@ public function ajaxRegrade($gradeable_id): array {
         return $this->core->getOutput()->renderResultMessage($msg, true);
     }
 
+    /**
+     * @return JsonResponse
+     * @param string $gradeable_id
+     */
+    #[Route("/api/{_semester}/{_course}/gradeable/{gradeable_id}/score", methods: ["GET"])]
+    public function ajaxGetGradeableScore(string $gradeable_id): JsonResponse {
+        // Instructor
+        if ($this->core->getUser()->getAccessLevel() !== 1) {
+            if (($_GET['user_id'] ?? '') !== $this->core->getUser()->getId()) {
+                return JsonResponse::getFailResponse('API key and specified user_id are not for the same user.');
+            }
+        }
+        try {
+            $gradeable = $this->core->getQueries()->getGradeableConfig($gradeable_id);
+        }
+        catch (\InvalidArgumentException $e) {
+            return JsonResponse::getFailResponse('Gradeable does not exist');
+        }
+        $gradeable_version_instance = $this->core->getQueries()->getGradedGradeable(
+            $gradeable,
+            $_GET['user_id'],
+            $gradeable->isTeamAssignment()
+        )->getAutoGradedGradeable()->getActiveVersionInstance();
+
+        if ($gradeable_version_instance === null || $gradeable_version_instance->isQueued() || $gradeable_version_instance->isGrading()) {
+            // Return null value for not graded, or not submitted gradeables.
+            return JsonResponse::getSuccessResponse(null);
+        }
```

Comment:
Would it make sense to return different values for each of these states, or should this endpoint be strictly for getting the score?  I'm fine with it either way, but I wanted to bring it up in case you hadn't thought about it.

Furthermore, why do we need an endpoint to get the score in the first place?  Would it not make sense to just have an endpoint which gets an entire gradeable, including the score and other metadata such as the grading status, grading time, etc?

Reply:
I will change it to get an array of the values instead of just score.